src/modules/blob.js

```diff
@@ -72,15 +74,31 @@ class BlobModule {
    * @return {object} The extrinsic to sign and send.
    */
   createNewTx(blob, keyPair = undefined, signature = undefined) {
+    let value = blob.blob;
+    if (!value) {
+      throw new Error('Blob must have a value!');
+    }
+
+    if (typeof value === 'object' && !Array.isArray(value) && !(value instanceof Uint8Array)) {
```

Comment:
I see what you are doing here but it would still be helpful to document what are you expecting and what conversions are you doing. 
I think we should accept `Uint8Array` or hex string only and keep the caller responsible for ensuring that. A dangerous example is caller passing the blob as an array with integers > 255. Polkadot's serialization will truncate the bigger integers to fit in a byte as the requires a byte array. So better to just check for `Uint8Array` and hex only

Reply:
fair enough observation, my initial idea with allowing normal arrays was just to prevent user from having to construct a typed array, but we can support only typed i dont think it'll make a huge difference for usability